sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveQl.scala

```diff
@@ -633,14 +633,28 @@ https://cwiki.apache.org/confluence/display/Hive/Enhanced+Aggregation%2C+Cube%2C
                    Token(script, Nil) ::
                    Token("TOK_SERDE", serdeClause) ::
                    Token("TOK_RECORDREADER", readerClause) ::
-                   outputClause :: Nil) :: Nil) =>
+                   outputClause) :: Nil) =>
 
+            // TODO the output should be bind with the output clause or RecordReader
             val output = outputClause match {
-              case Token("TOK_ALIASLIST", aliases) =>
+              case Token("TOK_ALIASLIST", aliases) :: Nil =>
                 aliases.map { case Token(name, Nil) => AttributeReference(name, StringType)() }
-              case Token("TOK_TABCOLLIST", attributes) =>
+              case Token("TOK_TABCOLLIST", attributes) :: Nil =>
                 attributes.map { case Token("TOK_TABCOL", Token(name, Nil) :: dataType :: Nil) =>
                   AttributeReference(name, nodeToDataType(dataType))() }
+              case Nil => // Not specified the output field names, let it be the same as input
+                (0 to inputExprs.length - 1).map { idx =>
+                  // Keep the same as Hive does, the first field names is "key", and second is
+                  // "value", however, Hive seems gives null string for the rest of the
```

Comment:
Thanks for notice that. I think this's probably a bug in Hive.
I did the queries in Hive CLI:
```
set hive.cli.print.header=true;
select transform(key + 1, key - 1, key) using '/bin/cat' from src limit 4;
```
![](https://raw.githubusercontent.com/chenghao-intel/githubimages/master/Selection_001.png)
```
create table test2 as select transform(key + 1, key - 1, key) using '/bin/cat' from src limit 4;
```
![](https://raw.githubusercontent.com/chenghao-intel/githubimages/master/Selection_002.png)
And print the result of the table `test2`:
![](https://raw.githubusercontent.com/chenghao-intel/githubimages/master/Selection_003.png)

You will see, it's not the expected result, of `key` and `value`, that's why I added the default field name for more than 2 columns.